tests/test_tables.py

```diff
@@ -1660,6 +1665,21 @@ def test_join_without_other_label(table, table2):
     2      | c      | 3     | two
     """)
 
+def test_join_with_empty_table(table, empty_table):
+    result = empty_table._join('points', table)
```

Comment:
While I think you're testing the right things - testing the internal methods (denoted with the underscore at the beginning) is not a standard practice. I think this could be a good PR if you did these same tests using the "join" method.